src/main/java/model/Product.java

```diff
@@ -11,17 +11,23 @@
 
     private String description;
     private int inventory;
+    private Long shopID;
+
+
+    private String name;
 
     @Lob
     private byte[] image;
 
     public Product() { }
 
-    public Product(String description, int inventory, byte[] image)
+    public Product(String name, String description, int inventory, byte[] image, Long shopID)
```

Comment:
Why store the shopId rather than just a reference to the shop?

Reply:
We store the shopId to keep code simple. So adding the reference would cause extra work that isn't needed.